src/components/FileDrop/FileDrop.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, useState } from "react"
+import { Column } from "../Layout"
+import { Text } from "../Text"
+
+import styles from "./FileDrop.module.scss"
+
+type Props = {
+  onFileChange: (file: File, hash: string) => void
+}
+
+const shortenFileName = (fileName: string) => {
```

Comment:
The problem with short filenames has been solved 💪
Although for filenames that just get truncated, the resulted filename is actually lengthened:

```node
> "abcdefghjijklm.pdf".length
18
> shortenFileName("abcdefghjijklm.pdf").length
23
> shortenFileName("abcdefghjijklm.pdf")
'abcdefgh...ghjijklm.pdf'
```